cmd/mountgen1.go

```diff
@@ -0,0 +1,243 @@
+   | ____|  |_____ | ____| | ____| |     |_____|  _____| |_____  |_____
+
+
+   Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+
+   Copyright © 2020-2022 Microsoft Corporation. All rights reserved.
+   Author : <blobfusedev@microsoft.com>
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package cmd
+
+import (
+	"blobfuse2/common"
+	"blobfuse2/common/config"
+	"blobfuse2/common/log"
+	"blobfuse2/component/azstorage"
+	"blobfuse2/component/file_cache"
+	"blobfuse2/component/libfuse"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var AzStorageOpt azstorage.AzStorageOptions
+var LibFuseOpt libfuse.LibfuseOptions
+var FileCacheOpt file_cache.FileCacheOptions
+var requiredFreeSpace int
+var configFile string
+var generateJsonOnly bool
+var gen1ConfigFilePath string
+
+var gen1Cmd = &cobra.Command{
+	Use:        "mountgen1",
+	Short:      "Mounts Azure Storage ADLS Gen 1 account using SPN auth",
+	Long:       "Mounts Azure Storage ADLS Gen 1 account using SPN auth",
+	SuggestFor: []string{"mntgen1", "gen1 mount"},
+	Args:       cobra.ExactArgs(1),
+	Hidden:     true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		options.MountPath = args[0]
+		options.ConfigFile = configFile
+		parseConfig()
+
+		err := config.Unmarshal(&options)
+		if err != nil {
+			fmt.Printf("Init error config unmarshall [%s]", err)
+			return err
+		}
+
+		err = options.validate(false)
+		if err != nil {
+			fmt.Printf("mountgen1: error invalid options [%v]", err)
+			return err
+		}
+
+		err = config.UnmarshalKey("azstorage", &AzStorageOpt)
+		if err != nil {
+			log.Err("mountgen1: AzStorage config error [invalid config attributes]")
+			return err
+		}
+
+		// not checking ClientSecret since adlsgen1fuse will be reading secret from env variable (ADL_CLIENT_SECRET)
+		if AzStorageOpt.AuthMode == "" && AzStorageOpt.ClientID != "" && AzStorageOpt.TenantID != "" {
```

Comment:
if client id, tenant-id are not given then we shall error out?

Reply:
Added the condition where if either among client id, tenant id or account name is empty, then we will return an error